.dotnet/src/Custom/Audio/AudioClient.cs

```diff
@@ -121,6 +121,54 @@ public virtual ClientResult<BinaryData> GenerateSpeechFromText(string text, Gene
         return ClientResult.FromValue(result.GetRawResponse().Content, result.GetRawResponse());
     }
 
+    /// <summary>
+    /// Generates text-to-speech audio using the specified voice speaking the provided input text, providing each transfer-encoded chunk as it arrives.
+    /// </summary>
+    /// <remarks>
+    /// <para> The result of this method may be iterated over using <c>await foreach</c>. </para>
+    /// The default format of the generated audio is <see cref="GeneratedSpeechFormat.Mp3"/> unless otherwise specified
+    /// via <see cref="SpeechGenerationOptions.ResponseFormat"/>.
+    /// </remarks>
+    /// <param name="text"> The text for the voice to speak. </param>
+    /// <param name="voice"> The voice to use. </param>
+    /// <param name="options"> Additional options to tailor the text-to-speech request. </param>
+    /// <returns> The generated audio in the specified output format. </returns>
+    public virtual AsyncResultCollection<BinaryData> GenerateSpeechFromTextStreamingAsync(string text, GeneratedSpeechVoice voice, SpeechGenerationOptions options = null)
```

Comment:
In my head, I was thinking this would return a `Stream` similar to how you can get the `ContentStream` when using the protocol method: https://github.com/openai/openai-dotnet/issues/45